File: trunk/PolarMathCS/PolarMath/Evaluation/Statistics/SDNN.cs
using PolarMath.Data;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PolarMath.Evaluation.Statistics
{
    public class SDNN : Evaluation<int>
    {
        public int evaluate(Training training) {
        int average = training.evaluate(new Average());
        long total = 0;
        foreach (int integer in training.getIntervals()) {
            total += (average - integer) * (average - integer);
        }
        return (int) Math.Sqrt(total / training.getIntervals().Count);
    }
    }
}

File: PolarMath/src/org/macondo/polar/evaluation/statistics/SDNN.java
package org.macondo.polar.evaluation.statistics;

import org.macondo.polar.data.Training;
import org.macondo.polar.evaluation.Evaluation;

/**
 * <p></p>
 *
 * Date: 16.03.2008
 *
 * @author <a href="mailto:ktsibriy@gmail.com">Kirill Y. Tsibriy</a>
 */
public class SDNN implements Evaluation<Integer>{
    public Integer evaluate(Training training) {
        Integer average = training.evaluate(new Average());
        long total = 0;
        for (Integer integer : training.getIntervals()) {
            total += (average - integer) * (average - integer);
        }
        return (int) Math.sqrt(total / training.getIntervals().size());
    }
}
